### PR TITLE
feature(cli): sync translated content

### PR DESCRIPTION
**Pull request by argl**
### Description

Implement sync-translated-content for the CLI
(MP-1610)

### Motivation

Ported over from yari

### CODE REVIEW

File: crates/rari-tools/src/sync_translated_content.rs [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+    Ok(())
+}
+
+fn md_or_html_exists(slug: &str, locale: Locale) -> Result<bool, ToolError> {
+    let folder_path = root_for_locale(locale)?
+        .join(locale.as_folder_str())
+        .join(slug.to_lowercase());
+    let md_path = folder_path.join("index.md");
+    let html_path = folder_path.join("index.html");
```
Comment (caugner) [inline crates/rari-tools/src/sync_translated_content.rs:329]: We no longer need to support `index.html` (cf. https://github.com/mdn/yari/pull/7575).

File: crates/rari-types/src/locale.rs [hunk 2]
```diff
@@ -112,6 +112,24 @@ static LOCALES_FOR_GENERICS_AND_SPAS: LazyLock<Vec<Locale>> = LazyLock::new(|| {
...
+    let translated_locales = [
+        Locale::Es,
+        Locale::Fr,
+        Locale::Ja,
+        Locale::Ko,
+        Locale::PtBr,
+        Locale::Ru,
+        Locale::ZhCn,
+        Locale::ZhTw,
+    ];
```
Comment (caugner) [inline crates/rari-types/src/locale.rs:125]:
```suggestion
    let translated_locales = [
        Locale::De,
        Locale::Es,
        Locale::Fr,
        Locale::Ja,
        Locale::Ko,
        Locale::PtBr,
        Locale::Ru,
        Locale::ZhCn,
        Locale::ZhTw,
    ];
```

**Review by caugner — CHANGES_REQUESTED** [review]
Only a quick review that doesn't replace @fiji-flo's.

File: crates/rari-types/src/locale.rs (hunk 2 above)
Comment (argl) [reply crates/rari-types/src/locale.rs:125]: Agreed, I was not aware that `de` is now officially supported

File: crates/rari-tools/src/sync_translated_content.rs (hunk 1 above)
Comment (argl) [reply crates/rari-tools/src/sync_translated_content.rs:329]: Ok, I'll remove that

**Review by fiji-flo — APPROVED** [review]
Looks good.

**Pull request merged**